### PR TITLE
feat(vault): migrate gitlab token

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Migrate gitlab token to vault

### Motivation
[ACIX-446](https://datadoghq.atlassian.net/browse/ACIX-446) and #incident-31294

### Describe how to test/QA your changes
Validate CI is green

### Additional Notes
Lint codeowners is [failing](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/666938922) and will be fixed by #29986 

[ACIX-446]: https://datadoghq.atlassian.net/browse/ACIX-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitlab/kernel_matrix_testing/common.yml [hunk 1]
```diff
@@ -200,7 +200,7 @@
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
   tags: ["arch:amd64"]
   before_script:
-    - GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $GITLAB_SCHEDULER_TOKEN) || exit $?; export GITLAB_TOKEN
+    - GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $GITLAB_TOKEN write_api) || exit $?; export GITLAB_TOKEN
```
Comment (brycekahle) [inline .gitlab/kernel_matrix_testing/common.yml:203]: I don't think we need the `write_api` token here, `read_api` should suffice. We are only reading from the Gitlab API.

**Review by spencergilbert — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

File: .gitlab/kernel_matrix_testing/common.yml (hunk 1 above)
Comment (chouetz) [reply .gitlab/kernel_matrix_testing/common.yml:203]: You are perfectly right, that was my initial intention and I mixed things up, sorry.

Comment (chouetz) [reply .gitlab/kernel_matrix_testing/common.yml:203]: Nice catch, as it was a `cleanup` I wrongly assumed you were doing write operations on gitlab.
Thanks!

**Review by CelianR — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Pull request merged**